ddl/ddl_api.go

```diff
@@ -2705,6 +2705,18 @@ func (d *ddl) ModifyColumn(ctx sessionctx.Context, ident ast.Ident, spec *ast.Al
 			if err := checkAutoIncrementRef(specNewColumn.Name.Name.L, dependColNames, t.Meta()); err != nil {
 				return errors.Trace(err)
 			}
+
+			// If there is an index on the generated column which we are trying to modify, return an error.
+			for _, idx := range t.Indices() {
+				if idx.Meta().State != model.StatePublic {
+					continue
+				}
```

Comment:
What happened if a write-only or delete-only state column is detected at this time?